app/webpacker/components/Delegates/DelegatesTable.jsx

```diff
@@ -1,53 +1,102 @@
 import React, { useMemo } from 'react';
-import { Button, Table } from 'semantic-ui-react';
+import { Button, Icon, Table } from 'semantic-ui-react';
 import _ from 'lodash';
 import cn from 'classnames';
+import { useSort, compareColumns } from './useSort';
 import I18n from '../../lib/i18n';
 import { competitionsUrl } from '../../lib/requests/routes.js.erb';
 import UserBadge from '../UserBadge';
 
 const dasherize = (string) => _.kebabCase(string);
 
+const COLUMN_TO_FIELD_PATH = {
+  name: 'user.name',
+  role: 'metadata.status',
+  region: 'metadata.location',
+  first_delegated: 'metadata.first_delegated',
+  last_delegated: 'metadata.last_delegated',
+  date_since_last_delegated: 'metadata.date_since_last_delegated',
+  total_delegated: 'metadata.total_delegated',
+};
```

Comment:
I like this idea of a path map that lets you use generalized comparator functions. I think if you combine this idea with our already existing `sortReducer`, it can become a very powerful framework for table data :)